scripts/completion.pl

```diff
@@ -141,15 +141,22 @@ sub parse_main_opts {
 sub trim { my $s = shift; $s =~ s/^\s+|\s+$//g; return $s };
 sub strip_dash { my $s = shift; $s =~ s/^-+//g; return $s };
 
-sub call_curl {
-    my ($cmd) = @_;
-    my $output = `"$curl" $cmd`;
-    if ($? == -1) {
-        die "Could not run curl: $!";
-    } elsif ((my $exit_code = $? >> 8) != 0) {
-        die "curl returned $exit_code with output:\n$output";
+sub read_source_files {
+    my @files = (
+        'src/tool_help.c',
```

Comment:
Have you actually tried this patch? If it produces any output at all, it's probably a coincidence. These files don't even contain the list of options, let alone in a format that would be a drop-in replacement for `curl --help all`. The source code that needs to be scanned are the files `docs/cmdline-opts/*` and only the relevant data needs to be extracted from them. Also, the path to the files needs to be relative to $(top_srcdir) so out-of-tree builds works.